File: src/main/java/ru/vmsystems/template/infrastructure/persistence/UserRepository.java
package ru.vmsystems.template.infrastructure.persistence;


import org.springframework.data.repository.CrudRepository;
import ru.vmsystems.template.domain.model.user.UserEntity;

import java.util.Optional;

public interface UserRepository extends CrudRepository<UserEntity, Long> {
    Optional<UserEntity> getById(Integer id);

    Optional<UserEntity> getByLogin(String login);

}

File: src/main/java/ru/vmsystems/glasscloud/user/UserRepository.kt
package ru.vmsystems.glasscloud.user


import org.springframework.data.repository.CrudRepository
import java.util.*

interface UserRepository : CrudRepository<UserEntity, UUID> {
    fun getById(id: UUID): UserEntity?
    fun getByLogin(login: String): UserEntity?
    fun getByCompanyId(companyId: UUID): List<UserEntity>
}
